src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/DefaultLSPDocumentMappingProvider.cs

```diff
@@ -58,5 +70,156 @@ public async override Task<RazorMapToDocumentRangeResponse> MapToDocumentRangeAs
 
             return documentMappingResponse;
         }
+
+        public async override Task<WorkspaceEdit> RemapWorkspaceEditAsync(WorkspaceEdit workspaceEdit, CancellationToken cancellationToken)
+        {
+            if (workspaceEdit == null)
+            {
+                return workspaceEdit;
+            }
+            else if (workspaceEdit.DocumentChanges != null)
+            {
+                // The LSP spec says, we should prefer `DocumentChanges` property over `Changes` if available.
+                var remappedEdits = await RemapVersionedDocumentEditsAsync(workspaceEdit.DocumentChanges, cancellationToken).ConfigureAwait(false);
+                return new WorkspaceEdit()
+                {
+                    DocumentChanges = remappedEdits
+                };
+            }
+            else
+            {
+                var remappedEdits = await RemapDocumentEditsAsync(workspaceEdit.Changes, cancellationToken).ConfigureAwait(false);
+                return new WorkspaceEdit()
+                {
+                    Changes = remappedEdits
+                };
+            }
+        }
+
+        private async Task<TextDocumentEdit[]> RemapVersionedDocumentEditsAsync(TextDocumentEdit[] documentEdits, CancellationToken cancellationToken)
+        {
+            var remappedDocumentEdits = new List<TextDocumentEdit>();
+            foreach (var entry in documentEdits)
+            {
+                var uri = entry.TextDocument.Uri;
+                var edits = entry.Edits;
+                if (!CanRemap(uri))
+                {
+                    // This location doesn't point to a background razor file. No need to remap.
+                    remappedDocumentEdits.Add(new TextDocumentEdit()
```

Comment:
Why rebuild the edit when we already have it available out of curiosity?

Reply:
Lol. This is what I get for refactoring it like 5 times 😆